{
  "cveId": "CVE-2024-41942",
  "title": "JupyterHub has a privilege escalation vulnerability with the `admin:users` scope",
  "description": "JupyterHub is software that allows one to create a multi-user server for Jupyter notebooks. Prior to versions 4.1.6 and 5.1.0, if a user is granted the `admin:users` scope, they may escalate their own privileges by making themselves a full admin user. The impact is relatively small in that `admin:users` is already an extremely privileged scope only granted to trusted users.\nIn effect, `admin:users` is equivalent to `admin=True`, which is not intended. Note that the change here only prevents escalation to the built-in JupyterHub admin role that has unrestricted permissions. It does not prevent users with e.g. `groups` permissions from granting themselves or other users permissions via group membership, which is intentional. Versions 4.1.6 and 5.1.0 fix this issue.",
  "publishedDate": "2024-08-08T14:36:44.498000+00:00",
  "lastModifiedDate": "2024-08-08T15:17:06.179000+00:00",
  "severity": "HIGH",
  "cvssMetrics": [
    {
      "version": "3.1",
      "vectorString": "CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H",
      "baseScore": 7.2,
      "baseSeverity": "HIGH"
    }
  ],
  "epss": {
    "score": 0.0024,
    "percentile": 46.284
  },
  "riskScore": 38,
  "exploitationStatus": "UNKNOWN",
  "affectedSystems": {
    "cpeMatches": [],
    "vendors": ["jupyterhub"],
    "products": ["jupyterhub"]
  },
  "references": [
    {
      "url": "https://github.com/jupyterhub/jupyterhub/security/advisories/GHSA-9x4q-3gxw-849f",
      "source": "https://github.com/jupyterhub/jupyterhub/security/advisories/GHSA-9x4q-3gxw-849f",
      "tags": ["x_refsource_CONFIRM"]
    },
    {
      "url": "https://github.com/jupyterhub/jupyterhub/commit/99e2720b0fc626cbeeca3c6337f917fdacfaa428",
      "source": "https://github.com/jupyterhub/jupyterhub/commit/99e2720b0fc626cbeeca3c6337f917fdacfaa428",
      "tags": ["x_refsource_MISC"]
    },
    {
      "url": "https://github.com/jupyterhub/jupyterhub/commit/ff2db557a85b6980f90c3158634bf924063ab8ba",
      "source": "https://github.com/jupyterhub/jupyterhub/commit/ff2db557a85b6980f90c3158634bf924063ab8ba",
      "tags": ["x_refsource_MISC"]
    }
  ],
  "attackTechniques": [],
  "metadata": {
    "tags": ["CWE-274"],
    "requiresUserInteraction": null,
    "requiresPrivileges": null,
    "attackVector": null,
    "sources": [
      {
        "name": "CVEList",
        "url": "https://github.com/CVEProject/cvelistV5/blob/main/cves/2024/41xxx/CVE-2024-41942.json",
        "lastModified": "2024-08-08T15:17:06.179000+00:00"
      }
    ]
  }
}
